aiohttp/resolver.py

```diff
@@ -9,9 +9,14 @@
 
 try:
     import aiodns
-
-    aiodns_default = hasattr(aiodns.DNSResolver, "getaddrinfo")
-except ImportError:
+    import pycares
+
+    # Only use aiodns if it has the getaddrinfo method and the
+    # underlying pycares library can locate DNS servers.
+    aiodns_default = hasattr(aiodns.DNSResolver, "getaddrinfo") and bool(
+        pycares.Channel().servers
```

Comment:
Might be cleaner to add a helper to aiodns instead. 

Ie aiodns.verify_available()